android/app/src/main/java/com/rnuilib/MainApplication.java

```diff
@@ -1,112 +1,122 @@
 package com.rnuilib;
 
+import android.content.Context;
+
 import com.facebook.react.PackageList;
 import com.facebook.react.ReactNativeHost;
 import com.facebook.react.ReactPackage;
+import com.facebook.react.ReactInstanceManager;
 import com.reactnativenavigation.NavigationApplication;
 import com.reactnativenavigation.react.NavigationReactNativeHost;
 import com.reactnativenavigation.react.ReactGateway;
 import com.wix.reactnativeuilib.UiLibPackageList;
 
+import java.lang.reflect.InvocationTargetException;
 import java.util.Arrays;
 import java.util.List;
 
 
 public class MainApplication extends NavigationApplication {
+    private final ReactNativeHost mReactNativeHost =
 
-    @Override
-    protected ReactGateway createReactGateway() {
-        ReactNativeHost host = new NavigationReactNativeHost(this, isDebug(), createAdditionalReactPackages()) {
-            @Override
-            protected String getJSMainModuleName() {
-                return "demo";
-            }
-        };
-        return new ReactGateway(this, isDebug(), host);
-    }
+            new NavigationReactNativeHost(this) {
+                @Override
+                public boolean getUseDeveloperSupport() {
+                    return BuildConfig.DEBUG;
+                }
+
+                @Override
+                protected List<ReactPackage> getPackages() {
+                    @SuppressWarnings("UnnecessaryLocalVariable")
+                    List<ReactPackage> packages = new PackageList(this).getPackages();
+                    // Packages that cannot be autolinked yet can be added manually here, for example:
+                    // packages.add(new MyReactNativePackage());
+                    // packages.addAll(new UiLibPackageList(this).getPackageList());
```

Comment:
@M-i-k-e-l 
I left this in comment cause it fail the build. 
Not sure exactly how to solve it, i'm sure you'll have an idea.